src/app/grid/grid-sorting-styling/grid-sorting-styling.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, OnInit, ViewChild } from "@angular/core";
+import { DefaultSortingStrategy, IgxGridComponent, IgxSelectComponent, SortingDirection } from "igniteui-angular";
+import { DATA } from "../../data/localData";
+
+enum TYPE {
+    SINGLE = "single",
+    MULTI = "multiple"
+}
+@Component({
+    selector: "grid-sorting-styling",
+    styleUrls: ["./grid-sorting-styling.component.scss"],
+    templateUrl: "grid-sorting-styling.component.html"
+})
+
+export class SortingStylingComponent implements OnInit {
+    @ViewChild("grid1", { read: IgxGridComponent, static: true })
+    public grid1: IgxGridComponent;
+
+    @ViewChild(IgxSelectComponent, { static: false })
+    public igxSelect: IgxSelectComponent;
+
+    public data: any[];
+    public sortingTypes = [{ name: "Multiple Sort", value: TYPE.MULTI }, { name: "Single Sort", value: TYPE.SINGLE }];
+    public currentSortingType: TYPE = TYPE.SINGLE;
+
+    constructor() {
+    }
+    public ngOnInit(): void {
+        this.data = DATA;
+        this.grid1.sortingExpressions = [
+            {
+                dir: SortingDirection.Asc, fieldName: "ProductName",
```

Comment:
There is no such field as `ProductName` in the data, you can set the initial sorted column to be `CategoryName`